models/perm/access/repo_permission.go

```diff
@@ -143,135 +143,162 @@ func (p *Permission) ColorFormat(s fmt.State) {
 	log.ColorFprintf(s, format, args...)
 }
 
-// GetUserRepoPermission returns the user permissions to the repository
-func GetUserRepoPermission(ctx context.Context, repo *repo_model.Repository, user *user_model.User) (perm Permission, err error) {
+// GetUserRepoPermissions returns the user permissions to the repositories
+func GetUserRepoPermissions(ctx context.Context, repos repo_model.RepositoryList, user *user_model.User) (perms map[int64]*Permission, err error) {
 	if log.IsTrace() {
 		defer func() {
 			if user == nil {
-				log.Trace("Permission Loaded for anonymous user in %-v:\nPermissions: %-+v",
-					repo,
-					perm)
+				log.Trace("Permissions Loaded for anonymous user in %-v:\nPermissions: %-+v",
```

Comment:
```suggestion
				log.Trace("Permissions loaded for anonymous user in %-v:\nPermissions: %-+v",
```
Interesting, this is also the first time I see `%-+v`, although it was already present in the legacy code…